src/randomizer.py

```diff
@@ -2006,47 +2013,79 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+randomization_passes: List[Callable] = [
+    perm_temp_for_expr,
+    perm_expand_expr,
+    perm_reorder_stmts,
+    perm_add_mask,
+    perm_xor_zero,
+    perm_cast_simple,
+    perm_refer_to_var,
+    perm_float_literal,
+    perm_randomize_internal_type,
+    perm_randomize_external_type,
+    perm_randomize_function_type,
+    perm_split_assignment,
+    perm_sameline,
+    perm_ins_block,
+    perm_struct_ref,
+    perm_empty_stmt,
+    perm_condition,
+    perm_mult_zero,
+    perm_dummy_comma_expr,
+    perm_add_self_assignment,
+    perm_commutative,
+    perm_add_sub,
+    perm_inequalities,
+    perm_compound_assignment,
+    perm_remove_ast,
+    perm_duplicate_assignment,
+    perm_chain_assignment,
+    perm_pad_var_decl,
+]
+
+
 class Randomizer:
-    def __init__(self, rng_seed: int) -> None:
+    def __init__(
+        self,
+        settings: Optional[Mapping[str, Any]],
```

Comment:
A little bit skeptical to passing in a whole unparsed dict like this. Maybe reasonable if we want to add further knobs to randomization in the future though, and not just weights. We do want it to be Mapping[str, object] still, and add type-checking accessors, like `json_prop` in net/core.py -- that function could be renamed and moved to helpers.

Reply:
I just finished refactoring it so the weight overrides are applied in main, and settings is no longer passed to randomizer

just randomization_weights: Mapping[str, float]